api/src/main/java/io/minio/MinioClient.java

```diff
@@ -2774,27 +2901,198 @@ public void putObject(String bucketName, String objectName, InputStream stream,
+   * @throws NoResponseException
+   *           upon no response from server
+   * @throws IOException
+   *           upon connection error
+   * @throws XmlPullParserException
+   *           upon parsing response xml
+   * @throws ErrorResponseException
+   *           upon unsuccessful execution
+   * @throws InternalException
+   *           upon internal library error
+   * @throws InvalidAlgorithmParameterException
+   *           upon wrong encryption algorithm used
+   * @throws BadPaddingException
+   *           upon incorrect padding in a block
+   * @throws IllegalBlockSizeException
+   *           upon incorrect block
+   * @throws NoSuchPaddingException
+   *           upon wrong padding type specified
+   *
+   * @see #putObject(String bucketName, String objectName, String fileName, String contentType)
+   * @see #putObject(String bucketName, String objectName, String fileName)
+   * @see #putObject(String bucketName, String objectName, InputStream stream, long size, String contentType)
+   *
+   */
+  public void putObject(String bucketName, String objectName, InputStream stream, long size, String contentType,
+      EncryptionMaterials encryptionMaterials) throws InvalidBucketNameException, NoSuchAlgorithmException,
+      InsufficientDataException, IOException, InvalidKeyException, NoResponseException, XmlPullParserException,
+      ErrorResponseException, InternalException, InvalidArgumentException, NoSuchPaddingException,
+      IllegalBlockSizeException, BadPaddingException, InvalidAlgorithmParameterException {
+
+    if (encryptionMaterials == null) {
+      throw new InvalidArgumentException("empty encryption properties not allowed");
+    }
+
+    // Generate symmetric 256 bit AES key for data encryption
+    KeyGenerator symKeyGenerator = KeyGenerator.getInstance("AES");
+    symKeyGenerator.init(256);
+    SecretKey dataEncryptionKey = symKeyGenerator.generateKey();
+
+    // Generate an iv to be used for data encryption
+    SecureRandom ivSeed = new SecureRandom();
+    byte[] iv = new byte[16];
+    ivSeed.nextBytes(iv);
+
+    // Get CipherStream after encryption of input stream
+    CipherInputStream cipherInputStream = encryptionMaterials.encryptInputStream(stream, dataEncryptionKey, iv);
+
+    // encrypt the plain data key using master key
+    byte[] plainDataKey = dataEncryptionKey.getEncoded();
+    byte[] encryptedDataKey = encryptionMaterials.encryptContentKeys(plainDataKey);
+
+    // Prepare data to be put to the object header
+    String encDataKey = new String(Base64.encodeBase64(encryptedDataKey), StandardCharsets.UTF_8);
+    String ivString = new String(Base64.encodeBase64(iv), StandardCharsets.UTF_8);
+
+    // Set the headermap with encryption related metadata
+    Map<String, String> headerMap = new HashMap<>();
+    headerMap.put("x-amz-meta-x-amz-key", encDataKey);
+    headerMap.put("x-amz-meta-x-amz-iv", ivString);
+    // matdesc is unused
+    headerMap.put("x-amz-meta-x-amz-matdesc", "");
```

Comment:
I always see this value as '{}', let's do the same so we give other clients what they expect.